src/editors.ts

```diff
@@ -0,0 +1,369 @@
+let monacoEditors: any[] = [];
+function monacoListener() {
+  (window as any).monaco.editor.onDidCreateEditor((e: any) => {
+    monacoEditors.push(e);
+  });
+}
+if ((window as any).monaco?.editor) {
+  monacoListener();
+} else {
+  // The following attempts to add a Monaco listener for apps that use the react-monaco-editor package
+  // We check if the monaco.editor module is loaded whenever nodes are added/removed from the ReactDOM
+  let mutationObserver = new MutationObserver((_mutations: any[], observer) => {
+    if ((window as any).monaco?.editor) {
+      monacoListener();
+      observer.disconnect();
+    }
+  });
+  let reactRootNode = document.querySelector(".react-root");
```

Comment:
this is neat! can we use the `ReactDOM` global object directly (e.g., `ReactDOM.findDOMNode(this)`)? I don't think React attaches a deterministic class name on the root element (that's up to the user of the library), but I think `ReactDOM` will have a reference to it. if not, we can add a few other common selectors here, like `#root` and `#app`, and shorten the timeout from 10s to 3–5s.

Reply:
Spent some time looking around at other sites using Monaco and sometimes they're not using React, but still dynamically loading Monaco (example: Google Colab). I made this section more generic -- now if `monaco.editor` is not immediately available, we just listen to all DOM changes (set the root node to `document`). I also shortened the timeout to 3s.